From 1a6fcc16a33970cce21703b67488a5880ab7c0a6 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:06 -0400
Subject: [scsi] hpsa: update raid offload status on device rescan

Message-id: <1410186734-19986-16-git-send-email-thenzl@redhat.com>
Patchwork-id: 92030
O-Subject: [RHEL6.6 e-stor PATCH 15/83] hpsa: update raid offload status on device rescan
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

When rescanning for logical drives, store information about whather
raid offload is enabled for each logical drive, and update the driver's
internal record of this.

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 250fb125ff94465a20429c4765f079aa142c9d80)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 08386d9..7f01442 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -805,6 +805,11 @@ static void hpsa_scsi_update_entry(struct ctlr_info *h, int hostno,
 
  /* Raid level changed. */
  h->dev[entry]->raid_level = new_entry->raid_level;
+
+ /* Raid offload parameters changed. */
+ h->dev[entry]->offload_config = new_entry->offload_config;
+ h->dev[entry]->offload_enabled = new_entry->offload_enabled;
+
  dev_info(&h->pdev->dev, "%s device c%db%dt%dl%d updated.\n",
   scsi_device_type(new_entry->devtype), hostno, new_entry->bus,
   new_entry->target, new_entry->lun);
@@ -925,6 +930,10 @@ static inline int device_updated(struct hpsa_scsi_dev_t *dev1,
   */
  if (dev1->raid_level != dev2->raid_level)
   return 1;
+ if (dev1->offload_config != dev2->offload_config)
+  return 1;
+ if (dev1->offload_enabled != dev2->offload_enabled)
+  return 1;
  return 0;
 }
 
-- 
1.7.1